src/metrology_apis/__init__.py

```diff
@@ -8,15 +8,67 @@
 __all__ = ["__version__", "Dimension", "Quantity", "Unit"]
 
 
+@runtime_checkable
+class MetrologyNamespace[Q: Quantity, U: Unit, D: Dimension](Protocol):
+
+    @staticmethod
+    def asdimension(obj: str | D) -> D: ...
+
+    @staticmethod
+    def asunit(obj) -> U[D]: ...
```

Comment:
```suggestion
    def asunit(obj: str | U) -> U[D]: ...
```

should this be type hinted?

Reply:
perhaps https://github.com/quantity-dev/metrology-apis/issues/11 to consider